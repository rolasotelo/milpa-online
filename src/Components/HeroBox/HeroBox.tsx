import React from "react";

interface Props {
  onClick: () => void;
}

const PlayTranslator = ["PLAY", "JUGAR", "HRÁT"];

const HeroBox = (props: Props) => {
  const PlayBabel = PlayTranslator[Math.floor(Math.random() * 3)];

  return (
    <div className="w-20.38rem md:mx-8 md:w-60rem h-35rem bg-play-background ring-8 ring-inset ring-black mt-6">
      <div className="flex flex-col justify-center items-center">
        <div>
          <button
            className="bg-button-blue w-52 h-24 px-8 mt-32 md:mt-56 focus:outline-none focus:bg-button-blue-pressed"
            onClick={props.onClick}
            style={{
              fontFamily: "goodlife-sans-condensed, sans-serif",
              fontWeight: 400,
              fontStyle: "normal",
              fontSize: "2.8rem",
            }}
          >
            {PlayBabel}
          </button>
        </div>

        <div className="w-72 md:w-11/12 h-80 md:h-52 bg-milpa-cardgame-mobile md:bg-milpa-cardgame"></div>
      </div>
    </div>
  );
};

export default HeroBox;
