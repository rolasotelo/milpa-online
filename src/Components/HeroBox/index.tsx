import React from "react";

interface Props {
  onClick: () => void;
  text: string;
}

const HeroBox = (props: Props) => {
  return (
    <div className="relative flex-col w-screen">
      <div className="relative w-20.38rem mx-auto max-w-100vw md:w-herobox-web h-herobox-web bg-milpaBlue-default ring-8 ring-offset-0 ring-milpaBlue-dark mt-5 md:mt-herobox-top-web">
        <div className="absolute inset-x-0 md:-top-94px bg-nopal-herobox mx-auto max-w-100vw w-20.38rem h-herobox-web md:w-nopal-herobox md:h-nopal-herobox"></div>
        <div className="absolute inset-x-0 top-170px bg-milpa-mobile md:bg-milpa-web mx-auto w-milpa-mobile h-milpa-mobile md:w-milpa-web md:h-milpa-web">
          <button
            className="z-10 absolute -top-64 md:-top-60 right-0 md:right-10 bg-button-blue w-52 h-24 mt-32 md:mt-56 focus:outline-none focus:bg-button-blue-pressed pl-6 pt-5 focus:pl-4 focus:pt-3"
            onClick={props.onClick}
            style={{
              fontFamily: "goodlife-serif, sans-serif",
              fontWeight: 400,
              fontStyle: "normal",
              fontSize: "2.25rem",
            }}
          >
            {props.text}
          </button>
        </div>
      </div>
      <div className="absolute inset-x-0 top-32  md:top-44 bg-mountains-herobox mx-auto max-w-100vw md:w-mountains h-mountains"></div>
    </div>
  );
};

export default HeroBox;
