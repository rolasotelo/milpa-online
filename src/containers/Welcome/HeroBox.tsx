import React, { ReactNode } from "react";

interface Props {
  onClick: () => void;
  text: string;
}

interface PropsWithChildren {
  children: ReactNode;
}

function BrushStrokes() {
  return (
    <>
      <div className="absolute -inset-y-20 -left-40 h-brush-pink-1 w-brush-pink-1 bg-brush-pink-1" />
      <div className="absolute -inset-y-8 -right-40 h-brush-pink-2 w-brush-pink-2 bg-brush-pink-2" />
    </>
  );
}

function Layout(props: PropsWithChildren) {
  const { children } = props;
  return (
    <div className="relative flex-col w-screen overflow-x-hidden">
      <div className="relative w-20.38rem mx-auto max-w-100vw md:w-herobox-web h-herobox-web mt-5 md:mt-herobox-top-web">
        <BrushStrokes />
        {children}
      </div>
    </div>
  );
}

function Frame(props: PropsWithChildren) {
  const { children } = props;
  return (
    <div className=" relative w-20.38rem mx-auto max-w-100vw md:w-herobox-web h-herobox-web bg-milpaBlue-default ring-8 ring-offset-0 ring-milpaBlue-dark mt-5 md:mt-herobox-top-web">
      <div className="absolute left-1 top-1 bg-symbol-corner-1 w-8 h-8" />
      <div className="absolute right-1 top-1 bg-symbol-corner-4 w-8 h-8" />
      {children}
    </div>
  );
}

function PlayButton(props: { onClick: () => void; text: string }) {
  const { onClick, text } = props;
  return (
    <button
      type="button"
      className="z-10 absolute -top-64 md:-top-60 right-0 md:right-10 bg-button-blue w-52 h-24 mt-32 md:mt-56 focus:outline-none focus:bg-button-blue-pressed pl-6 pt-5 focus:pl-4 focus:pt-3"
      onClick={onClick}
      style={{
        fontFamily: "goodlife-serif, sans-serif",
        fontWeight: 400,
        fontStyle: "normal",
        fontSize: "2.25rem",
      }}
    >
      {text}
    </button>
  );
}

function CallToAction(props: { onClick: () => void; text: string }) {
  const { onClick, text } = props;
  return (
    <>
      <div className="absolute inset-x-0 md:-top-94px bg-nopal-herobox mx-auto max-w-100vw w-20.38rem h-herobox-web md:w-nopal-herobox md:h-nopal-herobox" />
      <div className="absolute inset-x-0 top-170px bg-milpa-mobile md:bg-milpa-web mx-auto w-milpa-mobile h-milpa-mobile md:w-milpa-web md:h-milpa-web">
        <PlayButton onClick={onClick} text={text} />
      </div>
    </>
  );
}

function HeroBox(props: Props) {
  const { onClick, text } = props;

  return (
    <Layout>
      <Frame>
        <CallToAction onClick={onClick} text={text} />
      </Frame>
    </Layout>
  );
}

export default HeroBox;
