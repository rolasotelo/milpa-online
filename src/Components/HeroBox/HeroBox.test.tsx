import React from "react";
import renderer from "react-test-renderer";

import HeroBox from "./HeroBox";

describe("<HeroBox />", () => {
  const defaultProps = {
    onClick: () => {},
    text: "PLAY",
  };
  const wrapper = renderer.create(<HeroBox {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
