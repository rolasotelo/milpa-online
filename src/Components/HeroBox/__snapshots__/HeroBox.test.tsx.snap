// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroBox /> render 1`] = `
<div
  className="w-20.38rem md:mx-8 md:w-60rem h-35rem bg-play-background ring-8 ring-inset ring-black mt-6"
>
  <div
    className="flex flex-col justify-center items-center"
  >
    <div>
      <button
        className="bg-button-blue w-52 h-24 mt-32 md:mt-56 focus:outline-none focus:bg-button-blue-pressed pl-3 pb-2 focus:pl-2 focus:pb-1"
        onClick={[Function]}
        style={
          Object {
            "fontFamily": "goodlife-sans-condensed, sans-serif",
            "fontSize": "2.8rem",
            "fontStyle": "normal",
            "fontWeight": 400,
          }
        }
      >
        PLAY
      </button>
    </div>
    <div
      className="w-72 md:w-11/12 h-80 md:h-52 bg-milpa-cardgame-mobile md:bg-milpa-cardgame"
    />
  </div>
</div>
`;
